[vm] Remove raw address display

georgemitenkov opened the pull request:
## Description

When `Container`, `Locals`, `ContainerRef` values are displayed, raw address is printed. This is useless information and also error prone due to non-determinism.

## How Has This Been Tested?
<!--
- Please ensure that the functionality introduced by this change is well tested and verified to work as expected.
- Ensure tests cover both happy and unhappy paths.
- List and link relevant tests.
-->

## Key Areas to Review
<!--
- Identify any critical parts of the code that require special attention or understanding. Explain why these parts are crucial to the functionality or architecture of the project.
- Point out any areas where complex logic has been implemented. Provide a brief explanation of the logic and your approach to make it easier for reviewers to follow.
- Highlight any areas where you are particularly concerned or unsure about the code's impact on the change. This can include potential performance or security issues, or compatibility with existing features.
-->

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Move Compiler
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

<!-- Thank you for your contribution! -->

georgemitenkov commented:
* **#16549** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/16549?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/16549?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

vineethk reviewed (approved):
Could you briefly say something about testing this fix (manual is also okay)?

georgemitenkov commented:
@vineethk I removed all occurrences of using raw addresses of value pointers (locals + container), these are no longer printed out but there are no actual "tests" for this.

Something we can consider: have a transactional test that returns a struct so we see "container" being printed, but not sure if this is worth it or not, let me know.

vineethk commented:
> Something we can consider: have a transactional test that returns a struct so we see "container" being printed, but not sure if this is worth it or not, let me know.

Might be good to have such a test just so that something breaks if the representation is changed in a future PR.

georgemitenkov commented:
@vineethk let me add a test for that, this is also related to function value printing 👍

georgemitenkov commented:
@vineethk added a test here, but these displays in a raw format in transactional tests so I rebased https://github.com/aptos-labs/aptos-core/pull/16538 on top and added a test to print function value with struct, which uses this format and is good for test.
